### ISSUE TITLE
[Bugs] Issues I've encountered up until now

### ISSUE THREAD

**Issue by Oplexitie** [Techokami/SonicWorldsNext#6]
**Framework Build:** GitHub download 07/12/2022

**Godot Version:** 3.5.1

**First issue:**
The pause menu doesn't properly reset after a level restart and breaks after the reset.
I personally fixed this by forcing the game to unpause when restarting the level.
![image](https://user-images.githubusercontent.com/72892666/206276077-6f2e75b2-65e8-4546-ab31-12672f8d8e9f.png)

**Second issue:**
When playing Sonic & Tails, bumping up the monitor through a pink platform while Tails is also interacting with that monitor can result in funky monitor gravity and Sonic being pulled up. I couldn't trigger this with single characters.
![bugissue](https://user-images.githubusercontent.com/72892666/206277736-78006698-1229-4325-b8a9-ccf702689849.gif)

**Third issue:**
When playing with Tails (or Sonic & Tails), when Tails holds on to a moving hanger (one that moves up and down), after a bit he starts to vibrate and then falls off. This issue doesn't seem to affect Sonic or Knuckles. I didn't record footage for this one, since the vibration isn't visible on the GIF.

**Other issue I encountered:**
I'm putting this here since I don't know if this is an issue related to Godot, my computer or the framework. But when modifying the sound or music volume, it seems like some values are skipped, it works fine but it's weird.
![bugissue2](https://user-images.githubusercontent.com/72892666/206279761-5f19ccf7-0c12-47ae-a1d9-c192fe47c1f8.gif)
You can see here that 44 is skipped for some reason.

Anyways, I hope this helps, and I hope it doesn't bother anyone that I compiled all these bugs into one report. Have a nice day.

**Comment by Renhoex** [comment]
Thanks for the report, I've gone through and fixed the issues reported here and pushed it to the main branch.

**Comment by Oplexitie** [comment]
No problem, after testing the newest build, I can confirm that all the issues have been addressed.
Although there still seems to be a bit of weirdness when two characters bump under the monitors.
![bugissue2](https://user-images.githubusercontent.com/72892666/206668697-8b40ae66-ddff-43b3-b81a-9fa4a24e6a14.gif)
Anyways, except for what I just pointed out, it's all fixed, so I can close this thread if you want.
Btw, sorry for closing and opening back up this thread I misclicked.

**Comment by Oplexitie** [comment]
After re-testing the monitor bug in the latest Godot 4 version, I can confirm that the monitor shifting bug has been fixed. 

Altough, I encountered a new bug with the monitor ^^' ...
If you destroy a monitor that's been pushed up by Tails (so box in mid air) with Sonic, the destroyed monitor will fall through the level and ignore all collision. This is a pretty minor issue, since triggering this bug is quite difficult but I tought I'd still bring it.
